### DISCUSSION TITLE
RGB LED on ESP32S3 board

### DISCUSSION THREAD

**Discussion by A622266** [micropython/micropython#9661] [Hardware & Peripherals] [answered]
I am using an [ESP32-S3-DevKitM-1](https://docs.espressif.com/projects/esp-idf/en/latest/esp32s3/hw-reference/esp32s3/user-guide-devkitm-1.html), which has an RGB LED.  

I try to turn on this LED with the following:
```py
from machine import Pin
led = Pin(48, Pin.OUT, drive=Pin.DRIVE_2)
led.on()
print(led.value())
```
The LED does not light, but the printed value is 1.  Since it is an RGB LED, I would have thought 3 GPIOs would be needed, but the document just says it is driven by GPIO48.  The silkscreen on the board also says 'RGB@IO48' next to the LED.  Any advice?

**Comment by Lobo-T** [comment]
It's a digital addressable LED.
https://www.rose-lighting.com/wp-content/uploads/sites/53/2020/05/SK68XX-MINI-HS-REV.04-EN23535RGB-thick.pdf
Try the Neopixel driver: https://docs.micropython.org/en/latest/esp32/quickref.html#neopixel-and-apa106-driver

**Reply by A622266** [reply]
Thank for making me aware of this type of LED
I did try the Neopixel driver as follows:
```py
from machine import Pin
from neopixel import NeoPixel

pin = Pin(48, Pin.OUT)   # set GPIO48  to output to drive NeoPixel
neo = NeoPixel(pin, 1)   # create NeoPixel driver on GPIO48 for 1 pixel
neo[0] = (255, 255, 255) # set the first pixel to white
neo.write()              # write data to all pixels
r, g, b = neo[0]         # get first pixel colour
print(r, g, b)
```

Result of the print statement is (255, 255, 255), so the program appears to run.  However, the LED does not light.  
Do I make an error in the script, or do you think this SK68XXMINI-HS may not be compatible with the driver?

**Reply by Lobo-T** [reply]
It definitely works on all three of my ESP32-S3-DevKitC-1's.
According to the documentation both the DevKitC and DevKitM should have a SK68XXMINI-HS on GPIO48.  So it should work, you may have to consider that the LED on your board is simply broken.

**Comment by steveh127** [comment]
Nothing wrong with the code. It works just fine for me on the same module. I'm running the latest stable version of micropython using picocom to access the REPL.  Unusually on this module you can run both via USB0 or ACM0 and both work just fine to turn on the LED. What are you using to run the software?  Is micropython up to date?

**Reply by A622266** [reply]
Thanks for your input.  I'm using mpremote and ACM0.  Below is the print out when I do a soft reset, which shows the connection and MicroPython version number.  I think v1.19.1 is the latest, so it looks okay to me:

```
Connected to MicroPython at /dev/ttyACM0
Use Ctrl-] to exit this shell

MPY: soft reboot
MicroPython v1.19.1-544-g89b320737 on 2022-10-16; ESP32S3 module with ESP32S3
Type "help()" for more information.
```

**Comment by steveh127** [comment]
That all looks good but doesn't explain why the code doesn't work for you. I did wonder if your initial attempt had killed the rgb led. Settings GPIO 48 as you did initially does block any subsequent attempt to use the neopixel module but only for that session. It's back to working after a reboot.  You also don't need the Pin.OUT in your pin assignment for the neopixel but it still works. I have to agree with Lobo-t that it looks like you have a dead LED...

**Reply by A622266** [reply]
I tried hitting the reset button a few times on the board, and now it does work! Thanks for your help on this steveh127 and Lobo-t

**Comment by A622266** [comment, marked as answer]
Working Script:
```
from machine import Pin
from neopixel import NeoPixel

pin = Pin(48, Pin.OUT)   # set GPIO48  to output to drive NeoPixel
neo = NeoPixel(pin, 1)   # create NeoPixel driver on GPIO48 for 1 pixel
neo[0] = (255, 255, 255) # set the first pixel to white
neo.write()              # write data to all pixels
r, g, b = neo[0]         # get first pixel colour
```

**Comment by WolleKette** [comment]
Maybe someone is interested. I had the same problem with a N16R8 board V1.3 (with USB-C ports) where I couldn't control the LED. On a board Rev. V1.1(Micro-USB ports) it worked.
I found the solution in a arduino forum [here](https://forum.arduino.cc/t/esp32-s3-onboard-rgb-led/1198754/11). You have to close the RGB-named solder bridge next to the LED on the board.
